veps/vep-1.md

```diff
@@ -0,0 +1,108 @@
+
+### Example 2: Deprecating and removing an obsolete utility
+
+Version `3.xx` of Vitess supports a utility that has been made obsolete by recently added functionality:
+
+```
+vtdescribe: an obsolete utility
+```
+
+Vitess `4.xx` **SHOULD** invoke a warning when attempting to use this utility, with the utility completely removed in Vitess `5.xx`.
+
+_This change is backward compatible in Vitess `4.xx` because the utility can still be used. It is forwards compatible because attempting to use it will suggest the alternative functionality which is available._
+
+## Exclusions
+
+The following is not subject to the Backwards and Forwards Compatibility Promise:
+
+* Functionality which has been marked as experimental. As new features are developed, the Vitess authors may release it in a stable release but as experimental. As feedback is received, changes may be made to the feature without respecting the regular deprecation cycle associated with backwards and forwards compatibility.
+
+* Functionality limited to the source code and/or build environment. If the Vitess authors desired to make a change, such as upgrading the required Go version, or switching from `govendor` to Go modules, there is no requirement to ensure compatibility with older versions of Go. The promise is intended for user facing functionality.
+
+## Support Lifecycle
+
+Because the release cycle includes a new release every 12 weeks, most fixes will only be pushed to the master branch. An exception to this process will be made for high severity bugs, in which case a fix will be available in the weekly release for all currently supported versions.
+
+High severity bugs in this context include CVEs, data corruption, wrong results or outage inducing issues. The Vitess team **MAY** decide to only backport bugs with higher user impact.
+
+## Supported Upgrade Paths
+
+Because the _Backwards and Forwards Compatibility Promise_ is only between single major versions, it is not recommended to skip versions when upgrading. i.e.
+
+	To use _Example 1_ (from above): Before upgrading, a user will be able to execute the statement `SELECT 1` on their Vitess `3.xx` system without error. If the user were to upgrade directly to `5.xx` this query would now result in an error by default, with a configuration option available to temporarily permit the query again.
+
+	Had the user upgraded to Vitess `4.xx` first, they would have received deprecation warnings when executing the soon-to-be unsupported query.
+
+	Further, had the user upgraded directly from `3.xx` to `6.xx` there would be no longer be support for a compatibility configuration option. The query would just be unsupported.
+
+The Vitess team **MUST** document incompatibilities so that users who require skip-version upgrades can achieve this provided they read all release notes between versions.
```

Comment:
where will this be documented exactly? i'd advocate for having release notes that are linked from the main `vitess.io` site in a format that makes it easy for users to determine what changes they'll be pulling in when releasing vitess into production.

Reply:
The current release notes are [through github](https://github.com/vitessio/vitess/releases) releases. I was thinking of adding a section at the top called "Important Changes". I agree that we should cross-post to the docs. I have seen other projects do this.

Reply:
TIL about the github release notes. thanks for that!

I do think having them in the project themselves helps a great deal for discovery, especially for newcomers to the project.